docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+merge all of your separate resolver maps into one object before you pass it to
+`withClientState`.
+
+```js
+import merge from 'lodash.merge';
+import { withClientState } from 'apollo-link-state';
+
+import currentUser from './resolvers/user';
+import cameraRoll from './resolvers/camera';
+import networkStatus from './resolvers/camera';
+
+const stateLink = withClientState(
+  merge({}, currentUser, cameraRoll, networkStatus),
+);
+```
+
+<h2 id="cache">Updating the cache</h2>
+
+When you manage your local data with Apollo Client, your Apollo cache becomes
+the single source of truth for all your local & remote data. To update and read
+from the cache, you access it via the `context`, which is the third argument
+passed to your resolver function.
+
+<h3 id="write-data">writeData</h3>
+
+The easiest way to update the cache is with `cache.writeData`, which allows you
+to write data directly to the cache without passing in a query.
+`cache.writeData` is a method unique to `apollo-link-state`, so you won't be
+able to access it off your Apollo Client instance within your app. Here's how
+you use it in your resolver map for a simple update:
+
+```js
+const filter = {
+  Mutation: {
+    updateVisibilityFilter: (_, { visibilityFilter }, { cache }) => {
+      const data = { visibilityFilter, __typename: 'Filter' };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+};
+```
+
+`cache.writeData` also allows you to pass in an optional `id` property to write
+a fragment to an existing object in the cache. This is useful if you want to add
+some client-side fields to an existing object in the cache.
+
+The `id` should correspond to the object's cache key. If you're using the
+`InMemoryCache` and not overriding the `dataObjectFromId` config property, your
+cache key should be `__typename:id`.
+
+```js
+const user = {
+  Mutation: {
+    updateUserEmail: (_, { id, email }, { cache }) => {
+      const data = { email };
+      cache.writeData({ id: `User:${id}`, data });
+      return data;
+    },
+  },
+};
+```
+
+`cache.writeData` should cover most of your needs; however, there are some cases
+where the data you're writing to the cache depends on the data that's already
+there. In that scenario, you should use the DataProxy methods on the Apollo
+cache, which allow you to pass in a query or a fragment.
+
+<h2 id="cache-api">Cache API</h2>
+
+The [Apollo cache API](docs/react/features/caching.html) has several other
+methods to assist you with updating & retrieving data. Here are some common use
+cases where it's best to use the DataProxy methods instead of `cache.writeData`.
+
+<h3 id="write-query">writeQuery & readQuery</h3>
+
+Sometimes, the data you're writing to the cache depends on data that's already
+in the cache; for example, you're adding an item to a list or setting a property
+based on an existing property value. In that case, you should use
+`cache.readQuery` to pass in a query and read a value from the cache before you
```

Comment:
Wouldn't it be helpful to have a readData that takes an ID that's analogous to writeData?  Or is that basically what readFragment is (except without you having to specify a query)?

Reply:
readFragment is the counterpart that takes an ID, since queries always start at the root query object, which doesn't have an ID. that's maybe not a well known thing though

Reply:
well that doesn't ring true to me since write has writeData/writeQuery/writeFragment but read only has readQuery/readFragment?

Reply:
I think it's because it's possible to infer what you want when you do `writeData` without a query, but it's not really possible to know what data you want if we had a `readData` without a query.

Reply:
Oh haha, you're saying we can't just call `cache.readData()` and have it magically return the right data :)